server/bytestream/bytestream.go

```diff
@@ -53,28 +56,58 @@ func streamFromUrl(ctx context.Context, url *url.URL, grpcs bool, callback func(
 		return err
 	}
 	defer conn.Close()
-	client := bspb.NewByteStreamClient(conn)
 
-	// Request the file bytestream
-	req := &bspb.ReadRequest{
-		ResourceName: strings.TrimPrefix(url.RequestURI(), "/"), // trim leading "/"
-		ReadOffset:   0,                                         // started from the bottom now we here
-		ReadLimit:    0,                                         // no limit
-	}
-	readClient, err := client.Read(ctx, req)
-	if err != nil {
-		return err
-	}
+	if strings.Contains(url.String(), "/blobs/ac/") {
+		acClient := repb.NewActionCacheClient(conn)
+		instanceName, d, err := digest.ExtractDigestFromDownloadResourceName(strings.TrimPrefix(url.RequestURI(), "/"))
 
-	for {
-		rsp, err := readClient.Recv()
-		if err == io.EOF {
-			break
+		if err != nil {
+			return err
+		}
+		// Request the ActionResult
+		req := &repb.GetActionResultRequest{
+			InstanceName:      instanceName,
+			ActionDigest:      d,
+			InlineStdout:      true,
+			InlineStderr:      false,
+			InlineOutputFiles: []string{},
 		}
+		actionResult, err := acClient.GetActionResult(ctx, req)
 		if err != nil {
 			return err
 		}
-		callback(rsp.Data)
+
+		// Turn ActionResult into []byte
+		buf, err := proto.Marshal(actionResult)
+		if err != nil {
+			return err
+		}
+		callback(buf)
+
+	} else {
```

Comment:
in go, if you have some code like this:

if a {
  return "a"
} else {
 return "b"
}

it's common to write it as:
if a {
  return "a"
}
return "b"

And you could do that here -- basically just add "return nil" after callback(buf), and then remove the "} else {"